contracts/staking/cw20-stake/src/contract.rs

```diff
@@ -72,11 +72,17 @@ pub fn instantiate(
         None => None,
     };
 
+    let token_address = deps.api.addr_validate(&msg.token_address)?;
+    let _: TokenInfoResponse = deps
```

Comment:
What's going on here? Is this just a side-effecty thing? Would be nice to know what's happening.

I assume the point of this it to error if `InvalidCw20`, but I'm only able to infer that in the context of this PR. A comment here would be nice. It might also explain why this is necessary...

Reply:
thanks for the question. i've added a comment. this is a smoke test that we are dealing with a valid cw20. what's happening here:

1. we make a query against the provided address to see if it responds to the `TokenInfo` query message from the cw20 spec.
2. if the query fails, we transform the error into `InvalidCw20` and bail out.
3. if the query succeeds, we ignore the response as we don't actually care about the contents (hence, the `_` variable).